app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/dashboard/DashboardScreen.kt

```diff
@@ -118,8 +140,172 @@ private fun TabContent(
+            onClick = { changeCurrentProject(it) }
+        )
+
+        Spacer(Modifier.height(12.dp))
+    }
+}
+
+@OptIn(ExperimentalCoilApi::class)
+@Composable
+private fun ProjectCard(
+    project: Project,
+    isCurrent: Boolean,
+    onClick: () -> Unit
+) = Surface(
+    shape = shapes.medium,
+    border = if (isCurrent) {
+        BorderStroke(2.dp, MaterialTheme.colorScheme.primary)
+    } else {
+        BorderStroke(1.dp, MaterialTheme.colorScheme.outline)
+    },
+    modifier = Modifier
+        .fillMaxWidth()
+        .padding(horizontal = mainHorizontalScreenPadding, vertical = 4.dp)
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(
+                interactionSource = remember { MutableInteractionSource() },
+                indication = rememberRipple(),
+                onClick = onClick
+            )
+            .padding(16.dp)
+    ) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Image(
+                painter = rememberImagePainter(
+                    data = project.avatarUrl ?: R.drawable.default_avatar,
+                    builder = {
+                        error(R.drawable.default_avatar)
+                        crossfade(true)
+                    },
+                ),
+                contentDescription = null,
+                modifier = Modifier.size(46.dp)
+            )
+
+            Spacer(Modifier.width(8.dp))
+
+            Column {
+                Text(
+                    style = MaterialTheme.typography.bodySmall,
+                    color = MaterialTheme.colorScheme.primary,
+                    text = stringResource(
+                        when {
+                            project.isOwner -> R.string.project_owner
+                            project.isAdmin -> R.string.project_admin
+                            project.isMember -> R.string.project_member
+                            else -> 0
+                        }
+                    )
+                )
+
+                Spacer(Modifier.height(4.dp))
+
+                Text(
+                    text = project.name,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+            }
+        }
+
+        project.description?.let {
+            Spacer(Modifier.height(4.dp))
+            Text(
+                text = it,
+                style = MaterialTheme.typography.bodySmall
+            )
+        }
+
+        Spacer(Modifier.height(8.dp))
+
+        CompositionLocalProvider(
+            LocalContentColor provides MaterialTheme.colorScheme.outline
+        ) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                val iconSize = 18.dp
+                val indicatorsSpacing = 8.dp
+
+                @Composable
+                fun Indicator(@DrawableRes icon: Int, value: Int) {
```

Comment:
Wow, we can make description of composable in composable, looks very strange.... Why don't you want to move it, because this function is already too long?

Reply:
This function is only used inside this card component to make the code more concise and readable